### PR TITLE
Add support for pad to fusion translation

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds support for `pad` in FusionTranslation.

Details:
* Make pad accept dynamic vector in python_frontend

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/fusion_record.h [hunk 1]
```diff
@@ -442,114 +442,31 @@ struct ReshapeOpRecord : RecordFunctor {
...
+    const std::vector<Val*>& val_widths =
+        fd.getFusionStateVector(args_.at(1).index);
 
     TensorView* output = nullptr;
-    if (args_.at(1).stype == serde::StateType::Scalar) {
-      output = pad(arg, val_widths, fd.getFusionState(args_.at(1).index));
+    if (args_.at(2).stype == serde::StateType::Scalar) {
+      output = pad(arg, val_widths, fd.getFusionState(args_.at(2).index));
     } else { // default: None
       output = pad(arg, val_widths);
```
Comment (jjsjann123) [inline csrc/python_frontend/fusion_record.h:465]: nitpick: should we assert here?

File: csrc/python_frontend/translation.cpp [hunk 2]
```diff
@@ -863,6 +864,59 @@ class FusionTranslator : public OptInConstDispatch {
...
+    std::iota(normalized_indices.begin(), normalized_indices.end(), 0);
+
+    // Step 3: Transform to indices for original pad widths
+    std::vector<int64_t> original_indices;
+    original_indices.reserve(normalized_indices.size());
+    std::transform(
+        normalized_indices.begin(),
+        normalized_indices.end(),
+        std::back_inserter(original_indices),
+        [=](int64_t normalized_idx) {
```
Comment (jjsjann123) [inline csrc/python_frontend/translation.cpp:887]: This is just an for loop with irange?

Comment (jjsjann123) [reply csrc/python_frontend/translation.cpp:887]: sorry, this just reads a bit strange to me. Didn't mean to ask you to change it.

File: tests/python/test_python_frontend.py [hunk 3]
```diff
@@ -1810,6 +1810,27 @@ def fusion_func(fd: FusionDefinition):
...
+        def fusion_func(fd: FusionDefinition):
+            t0 = fd.from_pytorch(inputs[0])
+
+            S10 = fd.define_scalar(2.5, dtype=DataType.Float)
+            S13 = fd.define_scalar(7, dtype=DataType.Int)
+            S15 = fd.ops.mul(S10, S13)
+            S16 = fd.ops.cast(S15, dtype=DataType.Int)
+            V18 = fd.define_vector([S16, S16, S16, S16], dtype=DataType.Int)
+
+            t1 = fd.ops.pad(t0, V18)
```
Comment (jjsjann123) [inline tests/python/test_python_frontend.py:1827]: :clap: for new test on updated API.

**Review by jjsjann123 — APPROVED** [review]
LGTM. updated pad op record looks great.

File: csrc/python_frontend/translation.cpp (hunk 2 above)
Comment (rdspring1) [reply csrc/python_frontend/translation.cpp:887]: It is. Just having fun with `std::transform`

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
FYI, I disabled `shape_check` in `SequenceAsVector` for the pad op's shape because it supports negative values.

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Review by jacobhinkle — APPROVED** [review]
Nice to have dynamic pad in the frontend too!

**Pull request merged**